autocorpus/Autocorpus.py

```diff
@@ -378,17 +378,25 @@ def process_files(self):
             raise RuntimeError("A valid config file must be loaded.")
         # handle main_text
         if self.file_path:
-            soup = self.__handle_html(self.file_path, self.config)
-            self.main_text = self.__extract_text(soup, self.config)
-            try:
-                self.abbreviations = Abbreviations(
-                    self.main_text, soup, self.config, self.file_path
-                ).to_dict()
-            except Exception as e:
-                print(e)
+            file_extension = self.file_path.split(".")[-1].lower()
+            if file_extension in ["html", "htm"]:
+                soup = self.__handle_html(self.file_path, self.config)
+                self.main_text = self.__extract_text(soup, self.config)
+                try:
+                    self.abbreviations = Abbreviations(
+                        self.main_text, soup, self.config, self.file_path
+                    ).to_dict()
+                except Exception as e:
+                    print(e)
+            elif file_extension == "xml":
+                pass  # TODO: implement XML handling
```

Comment:
What should be done if the file is of neither type?

Reply:
good point, should raise an error in this situation

Reply:
The most appropriate error here is

```suggestion
                raise NotImplementedError("XML processing is not yet implemented for Auto-CORPus")
```